src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +152,1305 @@ static DamageTable D_80B65044 = {
+                    // make invisible by size:0
+                    Actor_SetScale(&randomFirePo->actor, 0);
+
+                    if (this->actor.child == NULL) {
+                        this->actor.child = &randomFirePo->actor;
+                    } else {
+                        randomFirePo->actor.child = this->actor.child;
+                        this->actor.child = &randomFirePo->actor;
+                    }
+                    break;
+                } else {
+                    randomIndex--;
+                }
+            }
+            enemyPtr = enemyPtr->next;
+        }
+    }
+
+    // remove unused fires
+    for (enemyPtr = FIRST_ENEMY; enemyPtr != NULL; enemyPtr = enemyPtr->next) {
+        if ((enemyPtr->id == ACTOR_EN_BIGPO) && (enemyPtr->params == ENBIGPO_POSSIBLEFIRE)) {
+            randomFirePo = (EnBigpo*)enemyPtr;
+            randomFirePo->actionFunc = EnBigpo_Die;
+            randomFirePo->actor.update = EnBigpo_UpdateFire;
+        }
+    }
+
+    EnBigpo_ChangeToFireCounting(this);
+}
+
+void EnBigpo_ChangeToFireCounting(EnBigpo* this) {
+    this->actionFunc = EnBigpo_FireCounting;
+}
+
+// count fires already found by dampe, once enough: spawn big poe for fight
+void EnBigpo_FireCounting(EnBigpo* this, GlobalContext* globalCtx) {
+    EnBigpo* firePo;
+    s32 activatedFireCount = 0;
+
+    for (firePo = (EnBigpo*)this->actor.child; firePo; firePo = (EnBigpo*)firePo->actor.child) {
+        if ((firePo->actor.params == ENBIGPO_REVEALEDFIRE) && (firePo->actionFunc == EnBigpo_RevealedFireIdle)) {
+            activatedFireCount++;
+        }
+    }
+
+    if (activatedFireCount == 3) { // all fires found
+        EnBigpo_SetupSpawn(this);
+    }
+}
+
+void EnBigpo_SetupFlameCirclePositions(EnBigpo* this, GlobalContext* globalCtx) {
+    EnBigpo* firePo;
+    Vec3f subCamEye;
+
+    this->idleTimer = 39;
+    for (firePo = (EnBigpo*)this->actor.child; firePo; firePo = (EnBigpo*)firePo->actor.child) {
+        EnBigpo_SetupFlameCircleCutscene(firePo);
+    }
+
+    // Setup sub camera: set 'eye' and 'at' and switch to subcamera
+    if (this->cutsceneSubCamId != SUBCAM_FREE) {
+        subCamEye.x = (Math_SinS(this->actor.yawTowardsPlayer) * 360.0f) + this->actor.world.pos.x;
+        subCamEye.y = this->actor.world.pos.y + 150.0f;
+        subCamEye.z = (Math_CosS(this->actor.yawTowardsPlayer) * 360.0f) + this->actor.world.pos.z;
+        Gameplay_CameraSetAtEye(globalCtx, this->cutsceneSubCamId, &this->actor.focus.pos, &subCamEye);
+    }
+
+    this->actionFunc = EnBigpo_DoNothing;
+}
+
+// nothing doing
+void EnBigpo_DoNothing(EnBigpo* this, GlobalContext* globalCtx) {
+}
+
+void EnBigpo_InitHiddenFire(EnBigpo* this) {
+    this->actor.draw = NULL;
+    this->actionFunc = EnBigpo_WaitingForDampe;
+}
+
+// idle until dampe turns ENBIGPO_POSSIBLEFIRE into ENBIGPO_REVEALEDFIRE
+void EnBigpo_WaitingForDampe(EnBigpo* this, GlobalContext* globalCtx) {
```

Comment:
That takes up so much more space and is uglier, is this for compatibility later on with another language?

Reply:
No clue what the original motivation for this style was. I think the idea is that for short comments, it doesn't look that good but it does look much better for multi-line comments, especially when there's lots of documentation needed to explain how a function works. And then it extends to be a consistent style above all functions. This discussion is more of a late-stage decomp discussion, so it's not a priority now and you can leave it as it was previously if you want

Reply:
This is just general style for function documentation in code bases. Function behaviour, arguments, and return values are described in block code. I believe it originated in Java called javadoc, but has since been used in other languages. We don't follow strictly to javadoc.